plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -5,13 +5,60 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 from __future__ import absolute_import, division, print_function
+
 __metaclass__ = type
 
 from ansible_collections.community.general.plugins.module_utils.redfish_utils import RedfishUtils
+from ansible.module_utils.basic import missing_required_lib
+
+import os
+import re
+import traceback
+
+HAS_IPADDRESS = True
+IPADDRESS_IMP_ERR = None
+try:
+    import ipaddress
+except ImportError as e:
+    IPADDRESS_IMP_ERR = traceback.format_exc()
+    HAS_IPADDRESS = False
+
+HAS_URLLIB3 = True
+URLLIB3_IMP_ERR = None
+try:
+    import urllib3
+except ImportError as e:
+    URLLIB3_IMP_ERR = traceback.format_exc()
+    HAS_URLLIB3 = False
+
+
+def ilo_certificate_login(root_uri, module, cert_file, key_file):
+    if not os.path.exists(cert_file):
+        module.fail_json(msg="The client cert file does not exist in the provided path %s" % str(cert_file))
+
+    if not os.path.exists(key_file):
+        module.fail_json(msg="The client key file does not exist in the provided path %s" % str(key_file))
+
+    try:
+        http = urllib3.PoolManager(cert_reqs='CERT_NONE', cert_file=cert_file, key_file=key_file)
+        cert_login = http.request('GET', root_uri + "/html/login_cert.html")
+    except Exception as e:
+        module.fail_json(msg="Server login with certificates failed: %s" % str(e))
+
+    return cert_login.getheader('X-Auth-Token')
 
 
 class iLORedfishUtils(RedfishUtils):
 
+    def __init__(self, creds, root_uri, timeout, module):
+        super().__init__(creds, root_uri, timeout, module)
+
+        if not HAS_IPADDRESS:
+            self.module.fail_json(msg=missing_required_lib('ipaddress'), exception=IPADDRESS_IMP_ERR)
+
+        if not HAS_URLLIB3:
+            self.module.fail_json(msg=missing_required_lib('urllib3'), exception=URLLIB3_IMP_ERR)
```

Comment:
These are breaking changes. The modules using this util worked so far when `ipaddress` and/or `urllib3` were not available, and now they will suddenly fail.

They should only fail when the user requests new features that need these libraries (when they are not around), but it should not fail for cases where the modules already worked in the past when these libraries were not available.